src/Auxiliary/Aux_CheckFileExist.cpp

```diff
@@ -30,17 +32,56 @@ bool Aux_CheckFileExist( const char *FileName )
 //
 // Note        :  Use the "stat" function to query the existence of the target folder
 //
-// Parameter   :  FileName : Name of the target folder
+// Parameter   :  FolderName : Name of the target folder
 //
-// Return      :  true/false <-> file exists/not exists
+// Return      :  true/false <-> folder exists/not exists
 //-------------------------------------------------------------------------------------------------------
 bool Aux_CheckFolderExist( const char *FolderName )
 {
 
    struct stat Buf;
 
-   if( stat( FolderName, &Buf ) != 0 )   return false; // not exist
-   else if( Buf.st_mode & S_IFDIR )      return true;  // is directory
-   else                                  return false; // not a diretory
+   if ( stat(FolderName, &Buf) != 0 )    return false; // not exist
+   else if ( !(Buf.st_mode & S_IFDIR) )  return false; // not a directoy
+
+   return true;
 
 } // FUNCTION : Aux_CheckFolderExist
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Aux_CheckPermission
+// Description :  Check whether or not the target file has requested permission
+//
+// Note        :  Use the "stat" function to query the permission of the target file
+//
+// Parameter   :  FileName : Name of the target file
+//             :  perms    : permission of the file (4->read, 2->write, 1->execute)
```

Comment:
IIUC, this routine supports checking a combination of multiple permissions (e.g., read+wrtie+execute=7). If so, clarify it and add an example.